overlord/snapstate/snapstate.go

```diff
@@ -563,14 +563,13 @@ func doInstall(st *state.State, snapst *SnapState, snapsup SnapSetup, compsups [
 		addTask(unlink)
 	}
 
-	// we need to know some of the characteristics of the device - it is
-	// expected to always have a model/device context at this point.
-	// TODO in a remodel this would use the old model, we need to fix this
-	// as needsKernelSetup needs to know the new model for UC2{0,2} -> UC24
-	// remodel case.
-	deviceCtx, err := DeviceCtx(st, nil, nil)
-	if err != nil {
-		return nil, err
+	if deviceCtx == nil {
```

Comment:
nitpick: you can pass this as the third argument below, it has an if like this inside

Reply:
Right, fixed now